hs_geographic_feature_resource/receivers.py

```diff
@@ -57,13 +60,22 @@ def check_fn_for_shp(files):
                 fileName, fileExtension = os.path.splitext(fullName.lower())
                 if ".shp" == fileExtension:
                     shp_filename = fileName
-                    shp = True
+                    if not shp:
+                        shp = True
+                    else:
+                        return False
```

Comment:
@hyi 
The else statement is to check the existence of any duplicate '.shp' file. The variable 'shp' is a global variable within this function and its value is False initially. Once a '.shp' file is detected, we set variable 'shp' to True. Since we just allow one '.shp' file in one feature res, in following rounds of this loop we check the value of variable "shp" to determine if there is any duplicate '.shp' file. If we find a '.shp' file, and the variable 'shp' is already True, that means this '.shp' file is duplicate as we set variable 'shp' to True in prior loop cycles. In this case we directly terminate the whole function by returning a False that means the files users uploaded are invalid (because we found duplicate ".shp" files in them).